### PR TITLE
Ghost name QoL changes

### LINKED ISSUE #17842: Ghost role naming and envy knife interaction
Free Golems and Ash Walkers have a random internal human name before they are renamed. This can occasionally persist when they die, meaning they have some random different name when they die.

Envy's knife also appears to steal this inner name rather than their external name.

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
rscadd: Ghosts now have the "Restore Character Name" verb, which will set
their ghost appearance and dead chat name to their character preferences.
bugfix: Mob spawners now give any generated names to the mind of the
spawned mob, meaning ghosts will have the name they had in life.
bugfix: Fixes interaction with envy's knife and lavaland spawns.
/:cl:

Fixes #17842.

### CODE REVIEW

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -590,6 +590,40 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		src << "<span class='notice'>Data HUDs enabled.</span>"
 		data_huds_on = 1
 
+/mob/dead/observer/verb/restore_ghost_apperance()
+	set name = "Restore Character Name"
+	set desc = "Sets your ghost name to the name of your roundstart character."
```
Comment (RemieRichards) [inline code/modules/mob/dead/observer/observer.dm]: >but it doesn't do just name

File: code/modules/mob/dead/observer/observer.dm [hunk 2]
```diff
@@ -590,6 +590,40 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		src << "<span class='notice'>Data HUDs enabled.</span>"
 		data_huds_on = 1
 
+/mob/dead/observer/verb/restore_ghost_apperance()
+	set name = "Restore Character Name"
```
Comment (RemieRichards) [inline code/modules/mob/dead/observer/observer.dm]: but it doesn't do just name

File: code/modules/mob/dead/observer/observer.dm [hunk 3]
```diff
@@ -590,6 +590,40 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
+	set desc = "Sets your ghost name to the name of your roundstart character."
+	set category = "Ghost"
+
+	set_ghost_appearance()
+
+/mob/dead/observer/proc/set_ghost_appearance()
+	if(!client)
+		world << "NO CLIENT"
```
Comment (RemieRichards) [inline code/modules/mob/dead/observer/observer.dm]: debug text

File: code/modules/mob/dead/observer/observer.dm [hunk 4]
```diff
@@ -590,6 +590,40 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
+		return
+	if(!client.prefs)
+		world << "NO PREFS"
```
Comment (RemieRichards) [inline code/modules/mob/dead/observer/observer.dm]: debug text

**Comment by optimumtact** [comment]
Feature freeze

**Comment by optimumtact** [comment]
Reopened with the understanding the ash walker tweak will be removed

File: code/modules/mob/dead/observer/observer.dm (hunk 2 above)
Comment (coiax) [inline code/modules/mob/dead/observer/observer.dm]: THINK OF THE GHOST HAIR

**Comment by Shadowlight213** [comment]
:-1: this is going to be a pain in the ass for administration, if someone says X griefed me, and admins look for x's ghost, but Surprise! X has changed his name to something completely different and now admins have to go dig up logs just to figure out who the they were.

**Comment by coiax** [comment]
@Shadowlight213 what about now? Only affects the name presented to deadchat, doesn't alter any of the vars of the ghost mob or mind.

**Comment by Shadowlight213** [comment]
better

**Pull request merged**